src/Microsoft.Azure.CosmosRepository/Specification/ISpecification.cs

```diff
@@ -19,20 +15,24 @@ public interface ISpecification<TItem, out TResult>
         /// <summary>
         /// The collection of filter expressions used for filtering queries.
         /// </summary>
-        IReadOnlyList<WhereExpressionInfo<TItem>> WhereExpressions { get; }
+        List<WhereExpressionInfo<TItem>> WhereExpressions { get; }
```

Comment:
I'm curious why the change from an `IReadOnlyList` to a `List`? This exposes the ability for consumers to call `.Add`, is that in fact desired?

Reply:
I did this for now until I understood whether or not this interface should even be `public` but I am coming to think it should not be.

Reply:
Updated this. Back to `IReadOnlyList<T>`/